scripts/ghidra/PatchestryDecompileFunctions.java

```diff
@@ -156,6 +203,167 @@ private HighVariable lValueOf(Varnode node) throws Exception {
         	return var;
         }
 
+        private String getDisplayName(DataType data_type) throws Exception {
+            return data_type.getDisplayName().replaceAll(" ", "");
+        }
+
+        private void serializePointerType(Pointer ptr) throws Exception {
+            name("name").value(getDisplayName(ptr));
+            name("kind").value("pointer");
+            name("size").value(ptr.getLength());
+            DataType elem_type = ptr.getDataType();
+            // Pointer element type could be null; assign void type to
+            // element in such case
+            if (elem_type == null) {
+                elem_type = VoidDataType.dataType;
+            }
+
+            name("element_type").value(label(elem_type));
+        }
+
+        private void serializeTypedefType(TypeDef typedef) throws Exception {
+            name("name").value(getDisplayName(typedef));
+            name("kind").value("typedef");
+            name("size").value(typedef.getLength());
+
+            DataType base_type = typedef.getBaseDataType();
+            // If base type is null, raise an exception.
+            // TODO(kumarak): if see cases where base type is null, handle it
+            // by assigning default `void` type.
+            if (base_type == null) {
+                throw new Exception("Typedef base type is null: " + typedef.toString());
+            }
+
+            name("base_type").value(label(base_type));
+        }
+
+        private void serializeArrayType(Array arr) throws Exception {
+            name("name").value(getDisplayName(arr));
+            name("kind").value("array");
+            name("size").value(arr.getLength());
+            name("num_elements").value(arr.getNumElements());
+            DataType elem_type = arr.getDataType();
+            if (elem_type != null) {
+                name("element_type").value(label(elem_type));
+            }
+        }
+            
+        private void serializeBuiltinType(DataType data_type, String kind) throws Exception {
+            name("name").value(getDisplayName(data_type));
+            name("size").value(data_type.getLength());
+            name("kind").value(kind);
+        }
+
+        private void serializeCompositeType(Composite data_type, String kind) throws Exception {
+            name("name").value(getDisplayName(data_type));
+            name("kind").value(kind);
+            name("size").value(data_type.getLength());
+            name("num_fields").value(data_type.getNumComponents());
```

Comment:
Don't need `num_fields` when there is an array of fields -- the array tells us how many fields there are.